src/server/worker/stats_test.go

```diff
@@ -249,3 +250,66 @@ func TestPrometheusStats(t *testing.T) {
 		})
 	}
 }
+
+// Regression: stats commits would not close when there were no input datums.
+//For more info, see github.com/pachyderm/pachyderm/issues/3337
+func TestCloseStatsCommitWithNoInputDatums(t *testing.T) {
+	c := getPachClient(t)
+	defer require.NoError(t, c.DeleteAll())
+
+	// TODO(ys): is this necessary?
```

Comment:
I'm actually not 100% sure either (I'll have to check), but I think it might be, and that would explain why we have so few stats tests. If so, we should probably put this stuff in its own testing harness someday (not in this PR :-))

Reply:
This is outdated, I checked and it is necessary